packages/mwp-api-proxy-plugin/src/util/send.js

```diff
@@ -82,54 +82,79 @@ function makeMockResponse(requestOpts, response = MOCK_RESPONSE_OK) {
 }
 
 /**
- * In order to receive cookies from `externalRequest` requests, this function
- * provides a cookie jar that is specific to the request.
+ * Function to build argument for `request` for API requests to DIY edge endpoints,
+ * i.e. endpoints served on a domain other than api.meetup.com
  *
- * The `requestUrl` is used to determine whether a cookie jar is needed
- *
- * https://github.com/request/request#examples
- *
- * @param {String} requestUrl the URL that will be used in the external request
- * @return {Object} a cookie jar compatible with the npm request `jar` API
+ * Currently, cookies cannot be set by DIY edge endpoints. (e.g. using the
+ * `jar` interface in `request`)
  */
-export const createCookieJar = requestUrl => {
-	const parsedUrl = url.parse(requestUrl);
-	if (parsedUrl.pathname === '/sessions') {
-		return externalRequest.jar(); // create request/url-specific cookie jar
-	}
-	return null;
+const buildDIYEdgeRequestArgs = externalRequestOpts => {
```

Comment:
I'm just slightly confused with naming -

* `DIY Edge` = non api.meetup.com
* `Shared Edge` = api.meetup.com (chapstick?)
* `External Request` = all requests, whether DIY or Shared?

I ask because I see that `buildDIYEdgeRequestArgs()` and `buildSharedEdgeRequestArgs()` both take an `externalRequestOpts` parameter. `external` to me tends to mean non-meetup endpoints, but it seems to refer all outbound requests? If that's the case, a nitpick suggestion would be to rename `externalRequestOpts` to simply `requestOpts`.

Reply:
Tangent question for my own knowledge - where do non meetup endpoints fall in this request structure? Are the handled separately elsewhere?

Reply:
Probably a bit more clear after discussing the DIY Edge definition, but 'external' is actually referring to 'outside of the API proxy server' - _any_ API request is an external request. That `externalRequestOpts` object was initially named to correspond to the `request` package, which is imported as `externalRequest` in this module - it's the module that makes the HTTP request from the proxy to the network. I wish it was named less ambiguously...

Reply:
> where do non meetup endpoints fall in this request structure? Are the handled separately elsewhere?

They're handled here - the proxy makes no assumptions about the responses other than that they are JSON-encoded, so non-meetup requests get mixed in with meetup requests in the query response array.